File: Latihan Kelas/KendaraanInheritance/Bus.cpp
#include "Bus.hpp"
// CONSTRUCTOR KELAS CHILD
// Default constructor untuk subclass Bus
Bus::Bus() : Kendaraan(){
    this->kapasitas = 0;
}

// User defined constructor untuk subclass Bus
Bus::Bus(int nomor, int tahun, string merk, int kapasitas) : Kendaraan(nomor, tahun, merk){
    this->kapasitas = kapasitas;
}

void Bus::printInfo(){
    Kendaraan::printInfo();
    cout << "Kategori: Bus" << endl;
}

int Bus::biayaSewa(int lamaSewa){
    return lamaSewa * 100000;
}

File: Latihan Kelas/KendaraanInheritance - Jawwa/Bus.java
public class Bus extends Kendaraan {
    private int kapasitas;

    public Bus() {
        super();
        this.kapasitas = 0;
    }

    public Bus(int nomor, int tahun, String merk, int kapasitas) {
        super(nomor, tahun, merk);
        this.kapasitas = kapasitas;
    }

    public Bus(Bus b) {
        super(b);
        this.kapasitas = b.kapasitas;
    }

    public void printInfo(){
        super.printInfo();
        System.out.println("Kategori: Bus");
        System.out.println("Kapasitas: " + kapasitas);
    }

    public int biayaSewa(int lamaSewa){
        return lamaSewa * 100000;
    }
}
